static/js/jira-configuration.js

```diff
@@ -324,7 +324,122 @@ $(".jiraConfiguration__info__backfillDate-label").each((_, backfillSinceLabelEle
+	const repoCountTd = $(self).children("td.repo-count");
+	const syncProgress = repoCountTd.children("span.inprogress-syncCount");
+	syncProgress.text(inprogressSyncStatus);
+};
+
+const updateBackfilledStatus = ({ subscriptionId, subscriptions, self }) => {
+	const syncCompleted = subscriptions[subscriptionId].syncCompleted;
+	const backfillSince = subscriptions[subscriptionId].backfillSince;
+	const repoStatusTd = $(self).children("td.repo-status");
+	const infoContainer = repoStatusTd.children(
+		`div.jiraConfiguration__infoContainer`
+	);
+	const inprogressIcon = infoContainer.children(
+		`div.jiraConfiguration__infoSpinner`
+	);
+	const syncStatusProgress = infoContainer.children(
+		`span.jiraConfiguration__table__syncStatus`
+	);
+	const backfillDateInfo = infoContainer.children('div.jiraConfiguration__info__backfillDate');
+	const backfillDate = infoContainer.children('span.jiraConfiguration__info__backfillDate-label');
+	let syncStatus = "IN PROGRESS";
+	if (syncCompleted) {
+		inprogressIcon.css("display", "none");
+		syncStatusProgress.removeClass("jiraConfiguration__table__in-progress");
+		syncStatusProgress.removeClass("jiraConfiguration__table__pending");
+		syncStatusProgress.addClass("jiraConfiguration__table__finished");
+		if(backfillSince){
+			backfillDate.text(backfillSince.toISOString());
+		}	
+		else{
+			backfillDateInfo.text("All commits backfilled");
+		}
+		syncStatus = "FINISHED";
+	}else{
+		syncStatusProgress.removeClass("jiraConfiguration__table__pending");
+		syncStatusProgress.addClass("jiraConfiguration__table__in-progress");
+	}
+	syncStatusProgress.text(syncStatus);
+};
+
+let fetchBackfillStateTimeout;
+function fetchAllConnectionsBackfillStatus() {
+	const subscriptionIds = getInprogressSubIds();
+	// check if we have atleast one subscription backfill in progress
+	if (subscriptionIds.length > 0) {
+		$.ajax({
+			type: "GET",
+			url: `/jira/subscriptions/backfill-status/?subscriptionIds=${subscriptionIds}`,
+			success: (response) => {
+				const data = response.data;
+				const subscriptions = data.subscriptions;
+				const backFillCompleted = data.backFillCompleted;
+
+				$(".jiraConfiguration__table__row").each(function () {
+					const self = this;
+					let subscriptionId = $(self).data("subscription-id");
+					if (subscriptionIds.includes(subscriptionId)) {
+						// repo count set
+						updateBackfilledRepoCount({ subscriptions, subscriptionId, self });
+						// repo status set
+						updateBackfilledStatus({ subscriptionId, subscriptions, self });
+					}
+				});
+				if (!backFillCompleted) {
+					fetchBackfillStateTimeout = setTimeout(
+						fetchAllConnectionsBackfillStatus,
+						6000
+					);
+				} else {
+					clearTimeout(fetchBackfillStateTimeout);
+				}
+			},
+			error: () => {
+				showErrorMessage(
+					"We couldn't fetch the backfill status of connections because something went wrong. Please try again."
```

Comment:
Actually if something wrong here, we can just do nothing. From UX point of view we don't need to notify user. We can simply retry in couple seconds automatically.